### ISSUE TITLE
Flagging Noisy Station -  Sprucehaven Receiver deployment: SpruceHaven (ID# 4360) Receiver ID: SG-F28CRPI33503

### ISSUE THREAD

**Issue by StuMackenzie** [MotusDev/Motus-TO-DO#610]
Flagging 

Receiver deployment: SpruceHaven (ID# 4360)
Receiver ID: SG-F28CRPI33503

As noisy and potential site for aliasing due to dense deployment of numerous tags:

All Purple Martin hits at this site follow a similar pattern to : https://motus.org/data/track?tagDeploymentId=16292

Also - Painted Buntings detected here are bogus.

Haven't looked through others.

**Comment by adamdsmith** [comment]
Do you want use this thread to flag all noisy stations? I can suggest some candidates for further review.

**Comment by StuMackenzie** [comment]
Yes, I thought we had a list going, but it makes sense to make a list here for further examination, repair where possible. Also perhaps termination of the station where warranted - eg. Formosa in Ontario.

**Comment by denislepage** [comment]
If not already there, it could be a good thing to add to the notification center. If there is a high number of run and hour bins being flagged with a high percentage of length 2. E.g. >100 runs and >80% of length 2, based on the activity table.

p.s. Do we actually know that 100 runs is a sufficiently high number to flag noisy periods? It doesn’t seem very high, and I wonder if Phil had looked at that aspect or if we only did pick a number ourselves.

**Comment by adamdsmith** [comment]
For starters:

Torrance C: Deployment 5521
Torrance B: 4777
GA_SSI_DOCK: 4303

I’ll add some others later. 

To @denislepage point, maybe we can use known noisy sites to put some bounds on “noisy periods,” or at least take a first stab at it, though the most obvious stations might be a biased sample.

**Comment by denislepage** [comment]
Another useful thing we could do in the new receiver dashboards is plot a percentage of runs that are length = 2 over time. We’d have to exclude periods with less than a certain amount of runs, so the sample is representative, or have a way to show the sample size visually as well as the percentage.

The highest number of runs we have in an hour so far is around 15,000.

**Comment by adamdsmith** [comment]
Adding:

- Johnson's Mills (deployment 5515) in September 2019
- Orchard Hill UMASS (deployments 4275 & 4284) in Sep-Oct 2017

**Comment by adamdsmith** [comment]
Something very odd happened in [mid-September on one of our stations](https://motus.org/data/receiver/deployment/plot/SG-3886RPI240C5-2019%20USFWS_R4%20Florida%20Panther%20NWR%2C%20FL.png?id=4844&n=100&firstTag=0). Looks like we detected a significant percentage of active tags in the entire network...

**Comment by jsayers-bsc** [comment]
> 
> 
> Something very odd happened in [mid-September on one of our stations](https://motus.org/data/receiver/deployment/plot/SG-3886RPI240C5-2019%20USFWS_R4%20Florida%20Panther%20NWR%2C%20FL.png?id=4844&n=100&firstTag=0). Looks like we detected a significant percentage of active tags in the entire network...

The other stations posted [here ](https://github.com/MotusDev/Motus-TO-DO/issues/674) are all very similar — a huge jump in false detections that lasted only a day or so. Many of the tags "detected" at each station are the same ones.

**Comment by adamdsmith** [comment]
How difficult would it be to track down the gz files for those station/days. I'd like to see what the event(s) that created these blasts of false positives looked like.

**Comment by denislepage** [comment]
If there is a list of station serial # and dates, that would be relatively simple to send you those files.

**Comment by adamdsmith** [comment]
10-4. I'll report back here with that list.

**Comment by adamdsmith** [comment]
I'm requesting a day either side of the event(s) for some noise context, hence the ranges...

CTT-867459049276348: 16-18 Jan 2020; 14-16 Feb 2020
SG-18B8RPI3A01C: 15-18 Feb 2020
SG-5113BBBK2668: 2-6 Sep 2019; 12-15 Sep 2019

**Comment by denislepage** [comment]
I’ll email you the link to download the files.

**Comment by adamdsmith** [comment]
See #674 for some summary/analysis of these requested stations/dates

**Comment by adamdsmith** [comment]
@denislepage If possible, I'd also be interested in the gz files for 2019-05-29 to 2019-06-28 for receiver SG-4001BBBK2170.

**Comment by StuMackenzie** [comment]
I'm not sure we talked about this explicitly, but as a starting point, one product that I'd like to see come of this is a tool to identify, or list, of a) noisy stations that we should talk to station owners about asap, and b) a tool or identify or list noisy antenna.

The solution for many of these stations is to take the station down, the alternative is to unplug or shift an antenna.

Point being, I don't think we should allow them to be collecting data and 'poisoning' our database ;)

**Comment by cjardine-bsc** [comment]
A query of the batches activity table would be what you need. 

Is this something internal you're looking for? I can set up a SQL query that Josh can access.

**Comment by StuMackenzie** [comment]
Yeah, wouldn't mind discussing what kind of threshold we can /should us to identify a station that is 'useless' vs one that's probably worth keeping. Same with antenna.

**Comment by StuMackenzie** [comment]
One crude flag may just be data file size in addition to more advanced methods.

**Comment by adamdsmith** [comment]
Having just vetted many detections, there were ~ 7 stations I simply ignored and tossed all possible detections. Here's how the hourly pulse counts from those stations compare to the rest of the stations I included. (NOTE, I excluded hours with extremely low and high pulse counts to make the chart more legible)

![image](https://user-images.githubusercontent.com/3195855/77200681-7311cc00-6ac1-11ea-82ad-f65026306c4a.png)

**Comment by StuMackenzie** [comment]
Some followup occurring in #aliasing discussion and #public data filters

**Comment by leberrigan** [comment]
I think this is better placed as a discussion since we're still trying to figure out the scope of the issue.

This is a network-wide problem that requires a network-wide solution. We can use single stations as examples to help identify noise problems, but eventually we'll want something that encompasses every possible source and mode (noise/aliasing).
